### PR TITLE
TW-570: Update sort for chat list by time and pin msg

### PR DESCRIPTION
**Pull request by nqhhdev**
### Issue: 
- #570 

### Resolved

https://github.com/linagora/twake-on-matrix/assets/99852347/accca145-93fc-4e7d-a5dd-57b46f76970f

https://github.com/linagora/twake-on-matrix/assets/99852347/b63fc225-f996-4c51-a8e2-8d28ae397a4a

### CODE REVIEW

**Review by drminh2807 — APPROVED** [review]

**Comment by drminh2807** [comment]
@nqhhdev  can you demo for Forward and Share flow. There are also have room list

File: lib/pages/chat_list/chat_list.dart [hunk 1]
```diff
@@ -109,8 +113,22 @@ class ChatListController extends State<ChatList>
...
+  }
+
+  int _sortListRoomByPinMessage(Room currentRoom, Room nextRoom) {
+    if (nextRoom.isFavourite && !currentRoom.isFavourite) {
+      return _ascendingOrder;
+    } else {
+      return _descendingOrder;
+    }
+  }
+
```
Comment (sherlockvn) [inline lib/pages/chat_list/chat_list.dart:131]: i think you can create a mixin

Comment (sherlockvn) [reply lib/pages/chat_list/chat_list.dart:131]: like `PresentationContactComparator`

Comment (nqhhdev) [reply lib/pages/chat_list/chat_list.dart]: I created a extension to handle it

**Review by drminh2807 — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**